### PR TITLE
Person and preservation record models

### PR DESCRIPTION
**Pull request by janid1967**
Person and PreservationRecord models w Migration
Tag: Moved props after ctors

### CODE REVIEW

File: src/Equinor.Procosys.Preservation.Domain/AggregateModels/TagAggregate/PreservationRecord.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            {
+                throw new ArgumentNullException(nameof(requirement));
+            }
+            if (timeService == null)
+            {
+                throw new ArgumentNullException(nameof(timeService));
+            }
+
+            RequirementId = requirement.Id;
+            NextDueTimeUtc = timeService.GetCurrentTimeUtc().AddDays(7*requirement.IntervalWeeks);
```
Comment (hbjorgo) [inline src/Equinor.Procosys.Preservation.Domain/AggregateModels/TagAggregate/PreservationRecord.cs]: Consider creating an extension method AddWeeks(this DateTime dateTime, int weeks).

File: src/Equinor.Procosys.Preservation.Domain/AggregateModels/TagAggregate/PreservationRecord.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        public bool? BulkPreserved { get; set; }
+        public DateTime? PreservedAtUtc { get; set; }
+        public int? PreservedBy { get; set; }
+        public string Comment { get; set; }
+ 
+        public void Preserve(Person preservedBy, string comment, ITimeService timeService) => Preserve(preservedBy, comment, false, timeService);
+
+        public void BulkPreserve(Person preservedBy, ITimeService timeService) => Preserve(preservedBy, null, true, timeService);
+
+        private void Preserve(Person preservedBy, string comment, bool bulkPreserve, ITimeService timeService)
```
Comment (hbjorgo) [inline src/Equinor.Procosys.Preservation.Domain/AggregateModels/TagAggregate/PreservationRecord.cs]: Would it be better to remove ITimeService and accept an argument DateTime preservedAtUtc?

File: src/Equinor.Procosys.Preservation.Domain/ITimeService.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace Equinor.Procosys.Preservation.Domain
```
Comment (hbjorgo) [inline src/Equinor.Procosys.Preservation.Domain/ITimeService.cs]: I think this doesn't belong here.

File: src/Equinor.Procosys.Preservation.Infrastructure/Entityconfigurations/PreservationRecordConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        public void Configure(EntityTypeBuilder<PreservationRecord> builder)
+        {
+            builder.Property(f => f.Schema)
+                .HasMaxLength(SchemaEntityBase.SchemaLengthMax)
+                .IsRequired();
+
+            builder.Property(x => x.Comment)
+                .HasMaxLength(PreservationRecord.CommentLengthMax);
+
+            builder.Property(x => x.PreservedAtUtc)
```
Comment (hbjorgo) [inline src/Equinor.Procosys.Preservation.Infrastructure/Entityconfigurations/PreservationRecordConfiguration.cs]: Is it possible to omit this configuration and use the defaults?

File: src/Equinor.Procosys.Preservation.Domain/AggregateModels/PersonAggregate/Person.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+        public Person(Guid oid, string firstName, string lastName) : base()
+        {
+            Oid = oid;
+            FirstName = firstName;
+            LastName = lastName;
+        }
+
+        public Guid Oid { get; private set; }
+        public string FirstName { get; private set; }
```
Comment (hbjorgo) [inline src/Equinor.Procosys.Preservation.Domain/AggregateModels/PersonAggregate/Person.cs]: Consider renaming to Firstname and Lastname.

Comment (janid1967) [reply src/Equinor.Procosys.Preservation.Domain/AggregateModels/PersonAggregate/Person.cs]: Won't fix

File: src/Equinor.Procosys.Preservation.Infrastructure/Entityconfigurations/PreservationRecordConfiguration.cs (hunk 4 above)
Comment (janid1967) [reply src/Equinor.Procosys.Preservation.Infrastructure/Entityconfigurations/PreservationRecordConfiguration.cs]: Removed config for nullable props. 
.IsRequired(false) is default behaviour for such

**Pull request merged**